src/modeler/AutoRest.Swagger/Validation/Extensions/XmsClientNameParameterValidation.cs

```diff
@@ -31,29 +32,29 @@ public class XmsClientNameParameterValidation : TypedRule<SwaggerParameter>
         /// <remarks>
         /// This may contain placeholders '{0}' for parameterized messages.
         /// </remarks>
-        public override string MessageTemplate => Resources.XmsClientNameInValid;
+        public override string MessageTemplate => Resources.XmsClientNameInvalid;
 
         /// <summary>
         /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
         /// </summary>
         public override Category Severity => Category.Error;
 
         /// <summary>
-        /// Validates if the name of property and x-ms-client-name(if exists) does not match.
+        /// Validates if the name of parameter model and x-ms-client-name(if exists) does not match.
         /// </summary>
-        /// <param name="definitions">Operation Definition to validate</param>
-        /// <returns>true if the validation succeeds. false otherwise.</returns>
-        public override bool IsValid(SwaggerParameter parameter)
+        /// <param name="parameter">Paramter model.</param>
+        /// <param name="context">Rule context.</param>
+        /// <returns></returns>
+        public override IEnumerable<ValidationMessage> GetValidationMessages(SwaggerParameter parameter, RuleContext context)
         {
-            if(parameter.Extensions != null && parameter.Extensions.Count != 0)
+            if (parameter?.Extensions?.Count != 0)
```

Comment:
This can be replaced by 
```
if(parameter?.Extensions?.ContainsKey(extensionToCheck)==true)
{
   if(parameter.Extensions[extensionToCheck]==parameter.Name)


```